test/test_formatter.rb

```diff
@@ -511,6 +511,18 @@ def test_format_between
     )
   end
 
+  def test_format_cast
+    assert_equals(
+      "should not add a space between colons in a cast",
+      strip_indent(
+        <<-EOB
+        screened_at :: DATE AS my_date
```

Comment:
Before this change, we would format this as `screened_at : : DATE AS my_date`, which is not valid.